src/pages/api/breadcrumbs.ts

```diff
@@ -135,54 +125,122 @@ async function fetchLabel(
+      };
     }
   }
   if (fieldName === 'taskId') {
     const task = await apiFetch(`/orgs/${orgId}/tasks/${fieldValue}`).then(
       (res) => res.json()
     );
-    return task.data.title;
+    return {
+      href: basePath + '/' + fieldValue,
+      label: task.data.title,
+    };
   }
   if (fieldName == 'viewId') {
     const view = await apiFetch(
       `/orgs/${orgId}/people/views/${fieldValue}`
     ).then((res) => res.json());
-    return view.data.title;
+    return {
+      folderId: view.data?.folder?.id,
+      href: basePath + '/' + fieldValue,
+      label: view.data.title,
+    };
   }
   if (fieldName == 'instanceId') {
     const journeyInstance = await apiFetch(
       `/orgs/${orgId}/journey_instances/${fieldValue}`
     ).then((res) => res.json());
-    return `${
-      journeyInstance.data.title || journeyInstance.data.journey.title
-    } #${journeyInstance.data.id}`;
+    return {
+      href: basePath + '/' + fieldValue,
+      label: `${
+        journeyInstance.data.title || journeyInstance.data.journey.title
+      } #${journeyInstance.data.id}`,
+    };
   }
   if (fieldName == 'journeyId') {
     const journey = await apiFetch(
       `/orgs/${orgId}/journeys/${fieldValue}`
     ).then((res) => res.json());
-    return journey.data.plural_label;
+    return {
+      href: basePath + '/' + fieldValue,
+      label: journey.data.plural_label,
+    };
   }
   if (fieldName == 'callAssId') {
     const assignment = await apiFetch(
       `/orgs/${orgId}/call_assignments/${fieldValue}`
     ).then((res) => res.json());
-    return assignment.data.title;
+    return {
+      href: basePath + '/' + fieldValue,
+      label: assignment.data.title,
+    };
   }
   if (fieldName == 'surveyId') {
     const survey = await apiFetch(`/orgs/${orgId}/surveys/${fieldValue}`).then(
       (res) => res.json()
     );
-    return survey.data.title;
+    return {
+      href: basePath + '/' + fieldValue,
+      label: survey.data.title,
+    };
   }
   if (fieldName == 'eventId') {
     const event = await apiFetch(`/orgs/${orgId}/actions/${fieldValue}`).then(
       (res) => res.json()
     );
-    return event.data.title;
+    return {
+      href: basePath + '/' + fieldValue,
+      label: event.data.title,
+    };
   }
-  return fieldValue;
+  if (fieldName == 'folderId') {
+    const folders = await apiFetch(`/orgs/${orgId}/people/view_folders`)
+      .then((res) => res.json())
+      .then((envelope) => envelope.data as ZetkinViewFolder[]);
+
+    const folder = folders.find((folder) => folder.id == parseInt(fieldValue));
+
+    if (folder) {
+      return {
+        folderId: `${folder?.parent?.id}`,
+        href: basePath + '/folders/' + fieldValue,
+        label: folder.title,
+      };
+    }
+  }
+
+  return null;
+}
+
+async function getFolderStack(
+  folderId: string,
+  basePath: string,
+  orgId: string,
+  apiFetch: ApiFetch
+) {
+  const folders = await apiFetch(`/orgs/${orgId}/people/view_folders`)
```

Comment:
Doesn't this mean that folders are retrieved twice? First by `fetchElement()` and then again by `getFolderStack()` in `fetchElements()` (plural). To me it seems like it should be possible to "expand" the stack of folders already after line 207 in your revision of this file. Will comment more there…